mod-source-record-manager-server/src/main/java/org/folio/dao/JobExecutionDaoImpl.java

```diff
@@ -42,166 +58,216 @@
       return jobExecutionFuture;
     }).compose(jobExecution -> {
-      CQLWrapper filter; //NOSONAR
+      CQLWrapper filter;
       try {
         filter = getCQLWrapper(TABLE_NAME, "id==" + jobExecution.getId());
       } catch (FieldException e) {
         throw new RuntimeException(e);
       }
-      Future<UpdateResult> updateHandler = Future.future(); //NOSONAR
-      pgClientFactory.createInstance(tenantId).update(tx, TABLE_NAME, jobExecution, filter, true, updateHandler);
-      return updateHandler;
+      Promise<UpdateResult> updateHandler = Promise.promise();
+      pgClientFactory.createInstance(tenantId).update(tx.future(), TABLE_NAME, jobExecution, filter, true, updateHandler);
+      return updateHandler.future();
     }).compose(updateHandler -> {
       if (updateHandler.getUpdated() != 1) {
         throw new NotFoundException(rollbackMessage);
       }
-      Future<Void> endTxFuture = Future.future(); //NOSONAR
-      pgClientFactory.createInstance(tenantId).endTx(tx, endTxFuture);
-      return endTxFuture;
+      Promise<Void> endTxFuture = Promise.promise();
+      pgClientFactory.createInstance(tenantId).endTx(tx.future(), endTxFuture);
+      return endTxFuture.future();
     }).setHandler(v -> {
       if (v.failed()) {
-        pgClientFactory.createInstance(tenantId).rollbackTx(tx, rollback -> future.fail(v.cause()));
+        pgClientFactory.createInstance(tenantId).rollbackTx(tx.future(), rollback -> promise.fail(v.cause()));
         return;
       }
-      future.complete(jobExecutionFuture.result());
+      promise.complete(jobExecutionFuture.result());
     });
-    return future;
+    return promise.future();
   }
 
   @Override
   public Future<Boolean> deleteJobExecutionById(String jobExecutionId, String tenantId) {
-    Future<UpdateResult> future = Future.future();
-    pgClientFactory.createInstance(tenantId).delete(TABLE_NAME, jobExecutionId, future.completer());
-    return future.map(updateResult -> updateResult.getUpdated() == 1);
+    Promise<UpdateResult> promise = Promise.promise();
+    pgClientFactory.createInstance(tenantId).delete(TABLE_NAME, jobExecutionId, promise);
+    return promise.future().map(updateResult -> updateResult.getUpdated() == 1);
+  }
+
+  private SqlSelect getSqlSelect(String tableName, String cql) throws FieldException {
+    try {
+      return new CQL2PgJSON(tableName + ".jsonb").toSql(cql);
+    } catch (QueryValidationException e) {
+      throw new CQLQueryValidationException(e);
+    }
+  }
+
+  private String prepareQueryGetJobWithoutParentMultiple(SqlSelect sqlSelect, int limit, int offset, String schemaName) {
+    String whereClause = String.format("WHERE %s ", sqlSelect.getWhere());
+    String orderBy = sqlSelect.getOrderBy().isEmpty() ? StringUtils.EMPTY : String.format("ORDER BY %s", sqlSelect.getOrderBy());
+    return String.format(getJobsWithoutParentMultipleSql, schemaName, whereClause, schemaName, schemaName, whereClause, orderBy, limit, offset);
```

Comment:
Do we really need "schemaName" 2 times in sequence as passing as arguments?

Reply:
yes, it will replace the corresponding placeholder in the query template